views/verify_email.py

```diff
@@ -0,0 +1,68 @@
+from rest_framework import generics, permissions
+from rest_framework.exceptions import NotFound
+from formula_one.serializers.generics.contact_information import ContactInformationSerializer
+from omniport.utils.switcher import load_serializer
+from omniport.settings.configuration.base import CONFIGURATION
+from settings.utils.verify_email import send_email, get_contact
+from rest_framework import generics, response, status
+from kernel.models import Person
+from base_auth.models import User
+from django.utils.http import urlsafe_base64_encode
+from django.utils.http import urlsafe_base64_decode
+from django.utils.encoding import force_bytes
+from django.contrib.auth.tokens import default_token_generator
+
+
+class VerifyEmailView(generics.GenericAPIView):
+    """
+    This view when responding to a GET request, generates email verification token
+    and sends mail to the concerned user.
+    """
+    permission_classes = [permissions.IsAuthenticated, ]
+    serializer_class = ContactInformationSerializer
+
+    def get(self, request):
+        """
+        View to serve GET requests
+        :param request: the request this is to be responded to
+        :return: the response for request
+        """
+        site_name = CONFIGURATION.site.nomenclature.verbose_name
+
+        url = 'http://stage.channeli.in/api/settings/verify_email/'
+        subject = f'{site_name} Email Verification'
+        body = f'To verify you email address, please visit {url}param'
+        user = request.user
+        contact = get_contact(user)
+        send_email(
+            uid=urlsafe_base64_encode(force_bytes(user.id)),
+            token=default_token_generator.make_token(request.user),
+            email_body=body,
+            email_subject=subject,
+            email_ids=[contact.email_address],
+            url=url,
+        )
+        return response.Response(
+            data=f'Email sent successfully to {contact.email_address}',
+            status=status.HTTP_200_OK,
+        )
+
+
+class VerifyTokenView(generics.GenericAPIView):
+    def get(self, request, *args, **kwargs):
+        """
+        View to serve POST requests to verify the url token and verify email upon validation
+        :param request: the request this is to be responded to
+        :return: the response for request
+        """
+        uidb64 = request.GET.get('uid')
+        token = request.GET.get('token')
+        user = User.objects.get(id=urlsafe_base64_decode(uidb64))
+        if_valid = default_token_generator.check_token(user, token)
+        if(if_valid):
+            Person.objects.get(user=user).contact_information.update(email_address_verified = True)
```

Comment:
Not sure if this is the right way. Since this will update `email_address_verfied` for **all** the `contact_information`'s instances unless we are sure we will maintain only **one** `contact_information` per `person`. @pradumangoyal what do you think?

Reply:
There is a big issue in the way we are verifying this, think of the situation I put my id and request for verification, the link will be sent to my email id. Then I change my email id to some other id and then click on the code on my email id, and so others will get verified. 

1.  Either we should disable the old link when contact information is updated
or
2. We should store the email id with the link or in the token instance